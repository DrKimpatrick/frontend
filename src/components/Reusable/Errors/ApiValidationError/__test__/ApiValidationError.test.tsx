import React from 'react';
import { cleanup, render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import Component from '../ApiValidationError';
import renderer from 'react-test-renderer';

const props = {
  fieldName: 'username',
  errors: [
    {
      username: 'username is required'
    }
  ]
};

describe('ApiValidationError', () => {
  afterEach(cleanup);

  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Component {...props} />, div);
  });

  it('should create component', () => {
    const tree = renderer.create(<Component {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should return empty when errors is not array', () => {
    const newProps = {
      ...props,
      errors: 'string'
    };
    const tree = render(<Component {...newProps} />);

    expect(tree).toBeTruthy();
  });
});
