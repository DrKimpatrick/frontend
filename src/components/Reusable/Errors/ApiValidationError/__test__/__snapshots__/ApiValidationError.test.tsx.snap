// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiValidationError should create component 1`] = `
<div
  className="inputError"
>
  username is required
</div>
`;
